### PR TITLE
Adding keymap for Nyquist

### PR DESCRIPTION
**Pull request by shovelpaw**
## Description
Added a keymap for the nyquist keyboard, which includes MIDI controls and mouse controls. 

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: keyboards/nyquist/keymaps/shovelpaw/config.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
+#define CONFIG_USER_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/config.h]: This line is not actually needed, and should be removed. 

Specifically, this should be included in the keyboard's `config.h` file, and is redundant. 

```suggestion
```

File: keyboards/nyquist/keymaps/shovelpaw/config.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/nyquist/keymaps/shovelpaw/config.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#define CONFIG_USER_H
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/config.h]:
```suggestion
```

File: keyboards/nyquist/keymaps/shovelpaw/config.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#define RGBLIGHT_VAL_STEP 8
+
+// Mouse specific settings
+#define MOUSEKEY_INTERVAL 16
+#define MOUSEKEY_DELAY 0
+#define MOUSEKEY_TIME_TO_MAX 60
+#define MOUSEKEY_MAX_SPEED 6
+#define MOUSEKEY_WHEEL_DELAY 0
+
+#endif
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/config.h]:
```suggestion
```

File: keyboards/nyquist/keymaps/shovelpaw/keymap.c [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  MIDI,
+  ADJUST
+};
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/nyquist/keymaps/shovelpaw/keymap.c [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+#define XXXXXXX KC_NO
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/keymap.c]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/nyquist/keymaps/shovelpaw/keymap.c [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+  _______, RESET  , RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI, RGB_SAD, KC_7,    KC_8,    KC_9,    _______, KC_DEL, \
+  _______, _______, _______, AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, KC_6,    KC_5,    KC_4,    _______, _______, \
+  _______, _______, _______, _______, _______, KC_LBRC, KC_RBRC, KC_1,    KC_2,    KC_3,    _______, _______, \
+  _______, _______, _______, _______, _______, KC_LCBR, KC_RCBR, KC_0,    KC_DOT,  _______, _______, _______ \
+)
+
+
+};
+
+#ifdef AUDIO_ENABLE
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/keymap.c]: This block isn't actually needed.  See below

File: keyboards/nyquist/keymaps/shovelpaw/keymap.c [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/keymap.c]: Could you remove this function?  

It's been depreciated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/nyquist/keymaps/shovelpaw/keymap.c [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_qwerty);
+        #endif
+        persistent_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/keymap.c]: This should be:
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

However, this entire macro can be removed, if you're only using QWERTY.

File: keyboards/nyquist/keymaps/shovelpaw/keymap.c [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+      if (record->event.pressed) {
+        layer_on(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+      break;
+   case MIDI:
```
Comment (drashna) [inline keyboards/nyquist/keymaps/shovelpaw/keymap.c]: This isn't needed.  `MO(_MIDI)` will accomplish this in your keymap.

Or you can add the following at the top of your keymap to accomplish the same thing. 
```c
#define MIDI MO(_MIDI)
```


The same applies to the ADJUST macro.

**Comment by shovelpaw** [comment]
@drashna requested changes have all been added. Thank you for reviewing!

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**